cmake/SetupPackages.cmake

```diff
@@ -81,3 +81,35 @@ endif ()
 if (CALIPER_FOUND)
     set(RT_LIBRARIES "${RT_LIBRARIES} ${caliper_LIB_DIR}/libcaliper.so" CACHE STRING "testing libraries" FORCE)
 endif()
+
+if (RAJA_ENABLE_TESTS)
+  include(ExternalProject)
+  # Set default ExternalProject root directory
+  SET_DIRECTORY_PROPERTIES(PROPERTIES EP_PREFIX ${CMAKE_BINARY_DIR}/tpl)
+
+  ExternalProject_Add(
+      googletest
+      GIT_REPOSITORY https://github.com/google/googletest.git
+      GIT_TAG release-1.7.0
+      INSTALL_COMMAND ""
+      LOG_DOWNLOAD ON
+      LOG_CONFIGURE ON
+      LOG_BUILD ON)
+
+  ExternalProject_Get_Property(googletest source_dir)
+  include_directories(${source_dir}/include)
+
+  ExternalProject_Get_Property(googletest binary_dir)
+  add_library(gtest      UNKNOWN IMPORTED)
+  add_library(gtest_main UNKNOWN IMPORTED)
+  set_target_properties(gtest PROPERTIES
+    IMPORTED_LOCATION ${binary_dir}/libgtest.a
+  )
+  set_target_properties(gtest_main PROPERTIES
+    IMPORTED_LOCATION ${binary_dir}/libgtest_main.a
+  )
+  add_dependencies(gtest      googletest)
```

Comment:
You've written it correctly, what I meant is that old versions of cmake 
(for like 5 years or some crazy thing) didn't actually honor those 
dependencies and tried to build in parallel right past them.  The output 
looked a bit like what that used to do, so it made me wonder about the 
cmake version on travis.

On 7 Jul 2016, at 12:34, David Beckingsale wrote:

> > +      LOG_CONFIGURE ON
>> +      LOG_BUILD ON)
>> +
>> +  ExternalProject_Get_Property(googletest source_dir)
>> +  include_directories(${source_dir}/include)
>> +
>> +  ExternalProject_Get_Property(googletest binary_dir)
>> +  add_library(gtest      UNKNOWN IMPORTED)
>> +  add_library(gtest_main UNKNOWN IMPORTED)
>> +  set_target_properties(gtest PROPERTIES
>> +    IMPORTED_LOCATION ${binary_dir}/libgtest.a
>> +  )
>> +  set_target_properties(gtest_main PROPERTIES
>> +    IMPORTED_LOCATION ${binary_dir}/libgtest_main.a
>> +  )
>> +  add_dependencies(gtest      googletest)
>
> @trws I think these dependencies should be enough to ensure the 
> correct build order for GoogleTest as an external project, but perhaps 
> not. I've added an explicit dependency of RAJA on GoogleTest which 
> should fix it.
>
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/LLNL/RAJA/pull/89/files/ee670308960fec80f5cb46523a3b8d233665c000#r69971919